### PR TITLE
blockdev_stream_remote_server_down: Remote server down when streaming

### PR DESCRIPTION
**Pull request by zhencliu**
Remote server down when doing block stream, job paused;
Resume remote storage, job should complete.

Signed-off-by: Zhenchao Liu <zhencliu@redhat.com>

ID: 1945100

### CODE REVIEW

**Comment by zhencliu** [comment]
virtio-scsi:
 (1/1) Host_RHEL.m8.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_stream_remote_server_down.q35: PASS (233.55 s)
virtio-blk:
 (1/1) Host_RHEL.m8.u3.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_stream_remote_server_down.q35: PASS (240.64 s)

**Comment by zhencliu** [comment]
Hi @ngu-niny @YongxueHong , can you please help to review this patch? Thanks.
@ngu-niny , I used a local qemu-nbd server to perform the test, the result is expected. Just fyi.

File: qemu/tests/blockdev_stream_remote_server_down.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+def run(test, params, env):
+    """
+    Suspend/resume remote storage service while doing block stream
+
+    test steps:
+        1. boot VM with a nbd image
+        2. format the data disk and mount it
+        3. create a file
+        4. add a snapshot image for data image
+        5. take snapshot on data image
```
Comment (ngu-niny) [inline qemu/tests/blockdev_stream_remote_server_down.py:76]: Should add the creating file step before do blockdev-stream.

File: qemu/tests/blockdev_stream_remote_server_down.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        6. do blockdev-stream
+        7. suspend nbd server
+        8. stream job should be paused
+        9. resume nbd server
+       10. wait till stream job completed
+       10. restart VM with snapshot disk, check all files and checksum
```
Comment (ngu-niny) [inline qemu/tests/blockdev_stream_remote_server_down.py:82]: Double step 10.

File: qemu/tests/blockdev_stream_remote_server_down.py (hunk 1 above)
Comment (zhencliu) [reply qemu/tests/blockdev_stream_remote_server_down.py:77]: Thank you Nini for pointing this out, I've revised the test steps, please have a check again.

File: qemu/tests/blockdev_stream_remote_server_down.py (hunk 2 above)
Comment (zhencliu) [reply qemu/tests/blockdev_stream_remote_server_down.py]: Updated.

**Comment by ngu-niny** [comment]
LGTM, test PASS:

 (1/1) Host_RHEL.m8.u3.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.1.s390x.io-github-autotest-qemu.blockdev_stream_remote_server_down.s390-virtio:  PASS (125.13 s)

Acked-by: Nini Gu <ngu@redhat.com>

**Review by YongxueHong — APPROVED** [review]
LGTM.

**Pull request merged**